lib/adagrams.rb

```diff
@@ -0,0 +1,218 @@
+# THIS PROGRAM WE WROTE PASSES 16/16 OF THE TESTS BUT DOES NOT FUNCTION PROPERLY IN THE TERMINAL. IF YOU UNCOMMENT LINE 61, THIS GAME RUNS PROPERLY IN THE TERMINAL BUT ONLY PASSES 15/16 TESTS.
+
+@words_created = []
+@final_winner = []
+
+@all_points = []
+in_progress = true
+
+
+while in_progress
+  in_progress = false
+
+  def draw_letters()
+    letters =
+    ["a", "a", "a", "a", "a", "a", "a", "a", "a", "b", "b", "c", "c", "d", "d", "d", "d", "e", "e", "e", "e", "e", "e", "e", "e", "e", "e", "e", "e", "f", "f", "g", "g", "g", "h", "h", "i", "i", "i", "i", "i", "i", "i", "i", "i", "j", "k", "l", "l", "l", "l", "m", "m", "n", "n", "n", "n", "n", "n", "o", "o", "o", "o", "o", "o", "o", "o", "p", "p", "q", "r", "r", "r", "r", "r", "r", "s", "s", "s", "s", "t", "t", "t", "t", "t", "t", "u", "u", "u", "u", "v", "v", "x", "y", "y", "z"]
+
+    pool = letters.map do |element|
+      element.upcase
+    end
+
+    pool = pool.shuffle
+
+    hand_drawn = []
+
+    10.times do |i|
+      hand_drawn << pool[i]
+    end
+
+    return hand_drawn
+  end
+
+
+  def uses_available_letters?(input, letters_drawn_input)
+    words_created = []
+
+    input_array = []
+    input.length.times do |i|
+      input_array << input[i]
+    end
+
+    testing_array = []
+    editted_letters_drawn = letters_drawn_input
```

Comment:
you are assigning the value of `letters_drawn_input` into a new local variable `editted_letters_drawn`. Is there a reason why you couldn't just keep using `letters_drawn_input`?